libs/sdk-ui-charts/src/highcharts/chartTypes/warterfallChart/waterfallChartsSeries.ts

```diff
@@ -0,0 +1,101 @@
+// (C) 2023 GoodData Corporation
+import { IMeasureGroupDescriptor } from "@gooddata/sdk-model";
+import { DataViewFacade } from "@gooddata/sdk-ui";
+import { IColorStrategy, valueWithEmptyHandling } from "@gooddata/sdk-ui-vis-commons";
+
+import { isOneOfTypes, parseValue, unwrap } from "../_util/common";
+import { multiMeasuresAlternatingTypes } from "../_chartOptions/chartCapabilities";
+import { IUnwrappedAttributeHeadersWithItems } from "../../typings/mess";
+import { IPointData, ISeriesItemConfig } from "../../typings/unsafe";
+
+function getSeriesItemData(
+    seriesItem: string[],
+    seriesIndex: number,
+    measureGroup: IMeasureGroupDescriptor["measureGroupHeader"],
+    viewByAttribute: IUnwrappedAttributeHeadersWithItems,
+    colorStrategy: IColorStrategy,
+    type: string,
+    emptyHeaderTitle: string,
+): IPointData[] {
+    return seriesItem.map((pointValue: string, pointIndex: number) => {
+        // by default seriesIndex corresponds to measureGroup label index
+        let measureIndex = seriesIndex;
+
+        if (isOneOfTypes(type, multiMeasuresAlternatingTypes) && !viewByAttribute) {
+            measureIndex = pointIndex;
+        }
+
+        const pointData: IPointData = Object.assign(
+            {
+                y: parseValue(pointValue),
+                format: unwrap(measureGroup.items[measureIndex]).format,
+            },
+            pointValue === null
+                ? {
+                      marker: {
+                          enabled: false,
+                      },
+                  }
+                : {},
+        );
+
+        pointData.name = valueWithEmptyHandling(
+            unwrap(measureGroup.items[measureIndex]).name,
+            emptyHeaderTitle,
+        );
+        pointData.legendIndex = pointIndex;
+        pointData.color = colorStrategy.getColorByIndex(pointData.y < 0 ? 2 : 1);
+
+        return pointData;
+    });
+}
+
+export function getWaterfallChartSeries(
+    dv: DataViewFacade,
+    measureGroup: IMeasureGroupDescriptor["measureGroupHeader"],
+    viewByAttribute: IUnwrappedAttributeHeadersWithItems,
+    type: string,
+    colorStrategy: IColorStrategy,
+    emptyHeaderTitle: string,
+) {
+    return dv
+        .rawData()
+        .twoDimData()
+        .map((seriesItem: string[], seriesIndex: number) => {
+            const seriesItemData = getSeriesItemData(
+                seriesItem,
+                seriesIndex,
+                measureGroup,
+                viewByAttribute,
+                colorStrategy,
+                type,
+                emptyHeaderTitle,
+            );
+
+            const seriesItemConfig: ISeriesItemConfig = {
+                upColor: colorStrategy.getColorByIndex(1), //Positive color,
+                color: colorStrategy.getColorByIndex(2), //Negative color,
```

Comment:
Possibly it would make more sense to me to use `color` (if we cannot rename it if this type is shared by other charts) as positive value and create new property for negative color `negativeColor`. `color` vs `upColor` seems bit cryptic to me and usually things should be centered about positive values.

Reply:
The upColor is one of attributes of the waterfall highchart. It is used for the positive value, and the color is used for the negative value. You can see here for more detail: https://api.highcharts.com/highcharts/plotOptions.waterfall.upColor. Thanks!